pyccel/codegen/printing/fcode.py

```diff
@@ -1319,43 +1332,29 @@ def _print_F2PYFunctionDef(self, expr):
                  imports,
                 'implicit none\n',
                  prelude,
+                 interfaces,
                  body_code,
                  'end {} {}\n'.format(func_type, name)]
 
         return '\n'.join(parts)
 
-    def _print_FunctionDef(self, expr):
-        self._handle_fortran_specific_a_prioris(list(expr.local_vars) +
-                                                list(expr.arguments)  +
-                                                list(expr.results))
-        # ... we don't print 'hidden' functions
-        if expr.hide:
-            return ''
-
-        name = self._print(expr.name)
-        self.set_current_function(name)
+    def _print_FunctionAddress(self, expr):
+        return expr.name
 
+    def function_signature(self, expr, name):
         is_pure      = expr.is_pure
         is_elemental = expr.is_elemental
-        is_procedure = expr.is_procedure
-
-        if expr.cls_name:
-            for k, m in list(_default_methods.items()):
-                name = name.replace(k, m)
-
-            cls_name = expr.cls_name
-            if not (cls_name == '__UNDEFINED__'):
-                name = '{0}_{1}'.format(cls_name, name)
-        else:
-            for i in _default_methods:
-                # because we may have a class Point with init: Point___init__
-                if i in name:
-                    name = name.replace(i, _default_methods[i])
-
         out_args = []
-        decs = OrderedDict()
         args_decs = OrderedDict()
-        # ... local variables declarations
+
+        for j, i in enumerate(expr.results):
+            if not i.name:
+                i.rename('out_{}'.format(j))
+        for j, i in enumerate(expr.arguments):
+            if not i.name:
+                i.rename('in_{}'.format(j))
```

Comment:
Why do we need to rename the variables?

Reply:
when a function is passed as an argument, we leave the name of it variables empty, as we only need them in case of fortran.
so these are just dummy variables of the prototype. we need to rename them so we can properly calculate `is_procedure`, they are also needed to print the call of a function argument with multiple returns.